libs/geoservices/covid/src/lib/pages/maps/components/lockdown-map/lockdown-map.component.ts

```diff
@@ -110,9 +110,8 @@ export class LockdownMapComponent implements OnInit {
   }
 
   public getStats(): void {
-    const statsUrl = 'https://nodes.geoservices.tamu.edu/api/covid/counties/stats';
-    const requests = forkJoin([this.http.get<DataRecord[]>(statsUrl)]);
-    requests.subscribe(([recievedData]) => {
+    const statService = this.countyService.getCountyStats();
+    const statSubscription = statService.subscribe((recievedData) => {
```

Comment:
Keeping the subscription to httpclient's observables is not necessary since they are self-completing.

Storing subscription references would be of some use if you'd want to manually unsubscribe from long-lived subscriptions at some point during or at the end of the component's life cycle. However, using the `takeUntil` operator to terminate the subscription is the preferred method.

You also  don't need to store the reference to the observable either, unless you plan to do things with it down the line, but since we're only interested in the value, that's not the case here.

So it should look like:

```js
this.countyService.getCountyStats().subscribe((receivedData) => {
   // Do stuff here.
})
```

Reply:
resolved